tests/ui/unseparated_prefix_literals.rs

```diff
@@ -24,7 +24,9 @@ fn main() {
     let _failf1 = 1.5f32;
     let _failf2 = 1f32;
 
+    // Test for macro
+    let _ = lit_from_macro!(); // Do NOT lint here |^
```

Comment:
You have to lint here, since that's how macro expansion works. The current behavior is totally fine :+1: